manager/orchestrator/update/updater.go

```diff
@@ -520,10 +532,45 @@ func (u *Updater) removeOldTasks(ctx context.Context, batch *store.Batch, remove
 	return removedTask, nil
 }
 
+// Checks if the current assigned node matches the Placement.Constraints
+// specified in the task spec for Updater.newService.
+func (u *Updater) nodeMatches(t *api.Task) bool {
+	if t.Spec.Placement == nil {
+		return true
+	}
+
+	constraints, _ := constraint.Parse(u.newService.Spec.Task.Placement.Constraints)
+
+	match := false
+	u.store.Update(func(tx store.Tx) error {
```

Comment:
I think this can be `View`. `Update` takes a lock and is only necessary if data is being written.